src/libsync/propagatecommonzsync.cpp

```diff
@@ -0,0 +1,320 @@
+{
+    ConfigFile cfg;
+    if (!propagator->account()->capabilities().zsyncSupported()) {
+        qCInfo(lcPropagator) << "[zsync disabled] Lack of server support.";
+        item->_isZsync = false;
+        return false;
+    }
+    if (item->_remotePerm.hasPermission(RemotePermissions::IsMounted) ||
+        item->_remotePerm.hasPermission(RemotePermissions::IsMountedSub)) {
+        qCInfo(lcPropagator) << "[zsync disabled] External storage not supported.";
+        item->_isZsync = false;
+        return false;
+    }
+    if (!(cfg.deltaSyncEnabled() ||
+          propagator->_deltaSyncEnabled.fetchAndAddAcquire(0))) {
+        qCInfo(lcPropagator) << "[zsync disabled] Client configuration option.";
+        item->_isZsync = false;
+        return false;
+    }
+    if (item->_size < cfg.deltaSyncMinFileSize() ||
+        item->_size < propagator->_deltaSyncMinFileSize.fetchAndAddAcquire(0)) {
+        qCInfo(lcPropagator) << "[zsync disabled] File size is smaller than minimum.";
+        item->_isZsync = false;
+        return false;
+    }
+
+    item->_isZsync = true;
+    return true;
+}
+
+void ZsyncSeedJob::seed()
+{
+    // Create a temporary file to use with zsync_begin()
+    QTemporaryFile zsyncControlFile;
+    zsyncControlFile.open();
+    zsyncControlFile.write(_zsyncData.constData(), _zsyncData.size());
+    zsyncControlFile.flush();
+
+    int fileHandle = zsyncControlFile.handle();
+    FILE* f = fdopen(dup(fileHandle), "r");
+    setvbuf(f, NULL, _IONBF, 0);
+    rewind(f);
+
+    struct zsync_state *zs;
+    if ((zs = zsync_begin(f)) == NULL) {
+        QString errorString = tr("Unable to parse zsync file.");
+        emit failedSignal(errorString);
+        return;
+    }
+
+    zsyncControlFile.close();
+    fclose(f);
+
+    std::string tmp = _zsyncFilePath.toStdString();
+    const char * fname = tmp.c_str();
+    {
+        /* Simple uncompressed file - open it */
+        FILE *f = fopen(fname, "r");
+        if (!f) {
+            QString errorString = tr("Unable to open file.");
+            emit failedSignal(errorString);
+            return;
+        }
+        else {
+
+            /* Give the contents to libzsync to read, to find any content that
+             * is part of the target file. */
+            qCInfo(lcZsyncSeed) << "Reading seed file:" << _zsyncFilePath;
+            zsync_submit_source_file(zs, f, false, _type == ZsyncMode::download ? false : true);
+
+            /* And close */
+            if (fclose(f) != 0) {
+                QString errorString = tr("Unable to close file.");
+                emit failedSignal(errorString);
+                return;
+            }
+        }
+    }
+
+    emit finishedSignal(zs);
+}
+
+/* stream_error(function, stream) - Exit with IO-related error message */
+void ZsyncGenerateJob::stream_error(const char *func, FILE * stream) {
+    qCWarning(lcZsyncGenerate).nospace() << func << ": " << strerror(ferror(stream));
+    deleteLater();
```

Comment:
I notice that `ZsyncGenerateJob` uses deleteLater while `ZsyncSeedJob` doesn't. I haven't seen the code using it yet, likely to introduce bugs!

I suggest you either parent these to something short-lived or deleteLater consistently. (we do both in some places)

Reply:
The reason is that `ZsyncGenerateJob` has a `QTemporaryFile` as part of the object, and I want that deleted. `ZsyncSeedJob` does not, I'm not completely sure how to parent to something 'short-lived'. If you still want me to add `deleteLater` to `ZsyncSeedJob`, I will.

Reply:
Done, moved to `QRunnable`.